config/config.go

```diff
@@ -235,3 +239,22 @@ type MetricsConfig struct {
 	// IgnoreErrors is a flag that instructs prometheus to ignore metric emission errors
 	IgnoreErrors bool `json:"ignoreErrors,omitempty"`
 }
+
+type WorkflowRequirements struct {
+	ReferenceOnly       bool `json:"referenceOnly"`
+	StrictReferenceOnly bool `json:"strictReferenceOnly"`
```

Comment:
what about `{reference: {strict:true}}`?

Reply:
@jessesuen and I settled on:

```yaml
data:
  workflowRestrictions:
    templateReferencing: Strict
```